### PR TITLE
STORM-3434: server: fix all checkstyle warnings

### PR DESCRIPTION
**Pull request by krichter722**

### CODE REVIEW

**Comment by srdo** [comment]
There's some clojure code referring to one of the renamed fields, which is why the test is failing here

**Comment by krichter722** [comment]
@srdo this can be merged, right?

**Comment by srdo** [comment]
I haven't looked through it yet, will do it in the next few days.

File: storm-server/src/main/java/org/apache/storm/LocalCluster.java [hunk 1]
```diff
@@ -462,6 +466,24 @@ public LocalTopology submitTopology(String topologyName, Map<String, Object> con
...
+            throws AlreadyAliveException, InvalidTopologyException, AuthorizationException, TException {
+        try {
+            @SuppressWarnings("unchecked")
+            Map<String, Object> conf = (Map<String, Object>) JSONValue.parseWithException(jsonConf);
+            submitTopology(name, conf, topology);
+        } catch (ParseException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
```
Comment (d2r) [inline storm-server/src/main/java/org/apache/storm/LocalCluster.java]: We've moved some code around here, but I also see we have suppressed a checkstyle warning. Are both of these changes necessary?

File: storm-server/src/main/java/org/apache/storm/Testing.java [hunk 2]
```diff
@@ -314,6 +308,21 @@ public static void withTrackedCluster(MkClusterParam param, TestJob code) {
...
+     * @param key the name of the metric to get.
+     * @return the metric
+     */
+    @SuppressWarnings("unchecked")
+    @Deprecated
+    public static int globalAmt(String id, String key) {
+        LOG.warn("Reading tracked metrics for ID {}", id);
+        return ((ConcurrentHashMap<String, AtomicInteger>) RegisteredGlobalState.getState(id)).get(key).get();
+    }
+
```
Comment (d2r) [inline storm-server/src/main/java/org/apache/storm/Testing.java]: Why was this moved?

File: storm-server/src/main/java/org/apache/storm/daemon/drpc/RequestFactory.java [hunk 3]
```diff
@@ -21,5 +21,6 @@
 import org.apache.storm.generated.DRPCRequest;
 
 public interface RequestFactory<T extends OutstandingRequest> {
-    public T mkRequest(String function, DRPCRequest req);
+
+    T mkRequest(String function, DRPCRequest req);
```
Comment (d2r) [inline storm-server/src/main/java/org/apache/storm/daemon/drpc/RequestFactory.java]: Is there a checkstyle policy that requires changes like these?

File: storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java [hunk 4]
```diff
@@ -1277,6 +1277,132 @@ private static void validatePortAvailable(Map<String, Object> conf) throws IOExc
...
+            }
+
+            if (Utils.exceptionCauseIsInstanceOf(InterruptedIOException.class, e)) {
+                throw e;
+            }
+            LOG.error("Error on initialization of nimbus", e);
+            Utils.exitProcess(13, "Error on initialization of nimbus");
+        }
+    }
+
```
Comment (d2r) [inline storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: FYI: This was moved and modified. The modifications were all white-space, except for reformatting one use of a ternary operator to put the `?` at the start of the following line.

File: storm-server/src/main/java/org/apache/storm/localizer/LocalizedResourceRetentionSet.java [hunk 5]
```diff
@@ -134,6 +134,7 @@ public String toString() {
         return "Cache: " + currentSize;
     }
 
+    @SuppressWarnings("checkstyle:AbbreviationAsWordInName")
```
Comment (d2r) [inline storm-server/src/main/java/org/apache/storm/localizer/LocalizedResourceRetentionSet.java]: I appreciate the judicious use of these suppressions.

File: storm-server/src/main/java/org/apache/storm/nimbus/AssignmentDistributionService.java [hunk 6]
```diff
@@ -186,7 +186,6 @@ private Integer nextQueueId() {
      * Get an assignments from the target queue with the specific index.
      * @param queueIndex index of the queue
      * @return an {@link NodeAssignments}
-     * @throws InterruptedException
```
Comment (d2r) [inline storm-server/src/main/java/org/apache/storm/nimbus/AssignmentDistributionService.java]: Why was this removed?

File: storm-server/src/main/java/org/apache/storm/pacemaker/PacemakerServer.java [hunk 7]
```diff
@@ -90,6 +90,7 @@ public PacemakerServer(IServerMessageHandler handler, Map<String, Object> config
         this.bossEventLoopGroup = new NioEventLoopGroup(1, bossFactory);
         // 0 means DEFAULT_EVENT_LOOP_THREADS
         // https://github.com/netty/netty/blob/netty-4.1.24.Final/transport/src/main/java/io/netty/channel/MultithreadEventLoopGroup.java#L40
+        int maxWorkers = (int) config.get(DaemonConfig.PACEMAKER_MAX_THREADS);
```
[the remark this answers is not in the document]
Comment (d2r) [reply storm-server/src/main/java/org/apache/storm/pacemaker/PacemakerServer.java]: I agree with @srdo on this one. If moving this up one line would cause a style violation, how about moving it down one line?

File: storm-server/src/main/java/org/apache/storm/scheduler/IsolationScheduler.java [hunk 8]
```diff
@@ -218,8 +218,8 @@ public void schedule(Topologies topologies, Cluster cluster) {
 
         List<ExecutorDetails> allExecutors = new ArrayList<ExecutorDetails>();
         Collection<List<ExecutorDetails>> values = compExecutors.values();
-        for (List<ExecutorDetails> eList : values) {
-            allExecutors.addAll(eList);
+        for (List<ExecutorDetails> value : values) {
+            allExecutors.addAll(value);
```
Comment (d2r) [inline storm-server/src/main/java/org/apache/storm/scheduler/IsolationScheduler.java]: What was the violation here?

File: storm-server/src/main/java/org/apache/storm/scheduler/multitenant/Node.java [hunk 9]
```diff
@@ -130,95 +133,110 @@ public static int countTotalSlotsAlive(Collection<Node> nodes) {
...
                     if (dontThrow) {
-                        LOG.warn("Worker slot [" + ws + "] can't be assigned to " + topId +
-                                 ". Its already assigned to " + topologySetEntry.getKey() + ".");
+                        LOG.warn("Worker slot ["
+                                + ws
+                                + "] can't be assigned to "
+                                + topId
+                                + ". Its already assigned to "
+                                + topologySetEntry.getKey()
+                                + ".");
```
Comment (d2r) [inline storm-server/src/main/java/org/apache/storm/scheduler/multitenant/Node.java]: This kind of enforced alignment will further encourage the use of format strings rather than string concatenation, because it is much less readable. I think this is a good thing.

I am not suggesting we change all such occurrences to use format strings now—if we see a performance issue or if we want to change it for readability later I consider it out of scope for this pull request.

File: storm-server/src/main/java/org/apache/storm/utils/ServerUtils.java [hunk 10]
```diff
@@ -214,13 +219,6 @@ public static ClientBlobStore getClientBlobStoreForSupervisor(Map<String, Object
 
     /**
      * Meant to be called only by the supervisor for stormjar/stormconf/stormcode files.
-     *
-     * @param key
-     * @param localFile
-     * @param cb
-     * @throws AuthorizationException
-     * @throws KeyNotFoundException
-     * @throws IOException
```
Comment (d2r) [inline storm-server/src/main/java/org/apache/storm/utils/ServerUtils.java]: Why are these removed?

**Comment by d2r** [comment]
```
[WARNING] Failure to transfer org.glassfish:javax.el:3.0.1-b07-SNAPSHOT/maven-metadata.xml from https://repository.apache.org/snapshots was cached in the local repository, resolution will not be reattempted until the update interval of apache.snapshots has elapsed or updates are forced. Original error: Could not transfer metadata org.glassfish:javax.el:3.0.1-b07-SNAPSHOT/maven-metadata.xml from/to apache.snapshots (https://repository.apache.org/snapshots): Connect to repository.apache.org:443 [repository.apache.org/207.244.88.140] failed: Connection timed out (Connection timed out)
```
This looks transient; I restarted the failed CI builds once.

File: storm-server/src/main/java/org/apache/storm/utils/ServerUtils.java (hunk 10 above)
Comment (krichter722) [reply storm-server/src/main/java/org/apache/storm/utils/ServerUtils.java]: They have have no description and thus no value. Leaving them empty is refused by checkstyle (which is a good thing imo).

File: storm-server/src/main/java/org/apache/storm/pacemaker/PacemakerServer.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (krichter722) [reply storm-server/src/main/java/org/apache/storm/pacemaker/PacemakerServer.java]: @d2r I don't get where the line `int maxWorkers` ought to be placed? Above the two comment lines, below or in between?

[the remark this answers is not in the document]
Comment (d2r) [reply storm-server/src/main/java/org/apache/storm/pacemaker/PacemakerServer.java]: I apologize, @krichter722 , it looked to me like the comment was resolved without a change, so I was confused by the diff. The code as it is currently looks fine to me.

File: storm-server/src/main/java/org/apache/storm/nimbus/AssignmentDistributionService.java (hunk 6 above)
Comment (d2r) [reply storm-server/src/main/java/org/apache/storm/nimbus/AssignmentDistributionService.java]: Answered [elsewhere](https://github.com/apache/storm/pull/3050#discussion_r312710620)

File: storm-server/src/main/java/org/apache/storm/LocalCluster.java (hunk 1 above)
Comment (krichter722) [reply storm-server/src/main/java/org/apache/storm/LocalCluster.java]: > We've moved some code around here

@d2r The reason is the check `OverloadMethodsDeclarationOrder` which is a good thing imo. The code moves are the outcome of the necessary rearrangement. Do you see an unnecessities?

> but I also see we have suppressed a checkstyle warning

Are you referrring to `@SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")`? It is necessary for the method to work properly, see @srdo's resolved review comments.

File: storm-server/src/main/java/org/apache/storm/Testing.java (hunk 2 above)
Comment (krichter722) [reply storm-server/src/main/java/org/apache/storm/Testing.java]: In order to keep the overloaded `withTrackedCluster` together

File: storm-server/src/main/java/org/apache/storm/daemon/drpc/RequestFactory.java (hunk 3 above)
Comment (krichter722) [reply storm-server/src/main/java/org/apache/storm/daemon/drpc/RequestFactory.java]: I'm 99.9% percent sure. I could verify if you want. In case it isn't I would make it a check because it's a redundant modifier.

File: storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java (hunk 4 above)
Comment (krichter722) [reply storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java]: Is now next to overloaded `launchServer`

File: storm-server/src/main/java/org/apache/storm/scheduler/IsolationScheduler.java (hunk 8 above)
Comment (krichter722) [reply storm-server/src/main/java/org/apache/storm/scheduler/IsolationScheduler.java]: `Local variable name 'eList' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'. [LocalVariableName]` which is good thing imo

**Comment by krichter722** [comment]
@d2r Thanks for you review. I think I covered everything, please notify me if I didn't - such large PRs are hard to oversee.

File: storm-server/src/main/java/org/apache/storm/LocalCluster.java (hunk 1 above)
Comment (d2r) [reply storm-server/src/main/java/org/apache/storm/LocalCluster.java]: Ah now I understand. This seems good to me.

File: storm-server/src/main/java/org/apache/storm/daemon/drpc/RequestFactory.java (hunk 3 above)
Comment (d2r) [reply storm-server/src/main/java/org/apache/storm/daemon/drpc/RequestFactory.java]: OK the first time I missed that `public` was removed as redundant. I had thought only a new line was added.

File: storm-server/src/main/java/org/apache/storm/scheduler/IsolationScheduler.java (hunk 8 above)
Comment (d2r) [reply storm-server/src/main/java/org/apache/storm/scheduler/IsolationScheduler.java]: OK, and it did not start with the minimum 2 lower-case letters. Thanks!

File: storm-server/src/main/java/org/apache/storm/daemon/drpc/RequestFactory.java (hunk 3 above)
Comment (krichter722) [reply storm-server/src/main/java/org/apache/storm/daemon/drpc/RequestFactory.java]: It's not required. It makes sense to remove these redundant modifiers. I'll open a separate PR for that.

**Comment by d2r** [comment]
@krichter722 Delayed merging for the 2.1.0 release branch. It looks like there is some conflict in one file.
@Ethanlm We would like to get this merged into master before to many more conflicts occur.
Are we OK to merge once the conflict is resolved?

File: storm-server/src/main/java/org/apache/storm/scheduler/multitenant/NodePool.java [hunk 11]
```diff
@@ -46,6 +46,12 @@ public static int slotsAvailable(NodePool[] pools) {
         return slotsAvailable;
     }
 
+    /**
+     * Get number of available slots.
+     * @return the number of nodes that are available to be taken
```
Comment (Ethanlm) [inline storm-server/src/main/java/org/apache/storm/scheduler/multitenant/NodePool.java]: This is `the number of slots` not `nodes`. But I am fine with it since you are just moving things around.

File: storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java [hunk 12]
```diff
@@ -439,26 +435,26 @@ protected String getWorkerLogWriter(SimpleVersion topoVersion) {
...
-            string = string.replace("%TOPOLOGY-ID%", _topologyId);
+            string = string.replace("%WORKER-ID%", workerId);
+            string = string.replace("%TOPOLOGY-ID%", topologyId);
             string = string.replace("%WORKER-PORT%", p);
             if (memOnheap > 0) {
                 string = string.replace("%HEAP-MEM%", String.valueOf(memOnheap));
             }
-            if (memoryLimitMB > 0) {
-                string = string.replace("%LIMIT-MEM%", String.valueOf(memoryLimitMB));
+            if (memoryLimitMb > 0) {
```
[the remark this answers is not in the document]
Comment (krichter722) [reply storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java]: @Ethanlm what does it stand for?

[the remark this answers is not in the document]
Comment (Ethanlm) [reply storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java]: It's Megabytes

**Comment by Ethanlm** [comment]
storm-server has 4 checkstyle violations. Probably due to recent merges. Could you please fix it? Thank you very much!

File: storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (krichter722) [reply storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java]: I read megabytes in your first post, but now I'm not getting why people would take it as megabits. I've never seen memory being measured in this unit, but megabytes or gigabytes instead. I wouldn't worry about it.

[the remark this answers is not in the document]
Comment (Ethanlm) [reply storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java]: Makes sense
